capstone/src/main/java/com/google/sps/servlets/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.sps.servlets;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.sql.DataSource;
+
+@WebListener("Creates a connection pool that is stored in the Servlet's context for later use.")
+public class ConnectionPoolContextListener implements ServletContextListener {
+
+    // Saving credentials in environment variables is convenient, but not secure - consider a more
+    // secure solution such as https://cloud.google.com/kms/ to help keep secrets safe.
+    private static final String CLOUD_SQL_CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+    private static final String DB_USER = System.getenv("DB_USER");
+    private static final String DB_PASS = System.getenv("DB_PASS");
+    private static final String DB_NAME = System.getenv("DB_NAME");
+
+    private DataSource createConnectionPool() {
+        // [START cloud_sql_postgres_servlet_create]
+        // The configuration object specifies behaviors for the connection pool.
```

Comment:
Let's remove all these comments, the code is easily readable without them. If you feel like you still want additional explanation consider better variable names instead, e.g. `config` -> `dbConfig`.